internal/auth/auth.go

```diff
@@ -68,6 +68,9 @@ func CreateAuthMiddleware(cfg *config.Config, log logrus.FieldLogger) (func(http
 		log.Println("k8s auth enabled")
 		authZ = K8sToK8sAuth{K8sAuthZ: authz.K8sAuthZ{ApiUrl: cfg.Auth.K8sApiUrl}}
 		authN = authn.OpenShiftAuthN{OpenShiftApiUrl: cfg.Auth.K8sApiUrl}
+	} else if cfg.Auth != nil && cfg.Auth.JwksUrl != "" {
+		log.Println("odic auth enabled")
+		authN = authn.JWTAuth{JwksUrl: cfg.Auth.JwksUrl}
```

Comment:
you need to define `authZ` too. Otherwise calls like this https://github.com/flightctl/flightctl/blob/main/internal/service/device.go#L45 will fail.
If we dont have actual permissions checks yet, the authZ can simply return true for everything (same as `NilAuth`)

Reply:
Authz should be decoupled from the authn interface. This will make it flexible to use different providers for authn and authz.

* auth: jwt token verifier
* authz: rebac or rbac
or
* authn: openshift 
* authz: opa or rebac

For now, I can add check call to return true

Reply:
I agree, and the interfaces are already decoupled. Or is something missing ?

Reply:
The interface is decoupled; I should have checked the nil_auth properly after the new updates. That's why I was wondering whether I could use NilAuth as authz and keep authz to jwtauth. 
```
authZ = NilAuth{}
authN = authn.JWTAuth{JwksUrl: cfg.Auth.JwksUrl, OidcDiscoveryUrl: cfg.Auth.OidcDiscoveryUrl}
```